### PR TITLE
Add read-only support for ACK resources via annotation

### PR DESCRIPTION
**Pull request by a-hilaly**
This commit adds support for making ACK resources read-only via an
annotation. A new `services.k8s.aws/read-only` annotation can be
added to a resource to instruct the ACK service controller to not
create/patch/delete the underlying AWS resource.

If the annotation is set to true, the controller will:
- return an error on create if the resource doesn't exist
- only patch metadata/spec for updates but not make cloud API calls
- skip deletion of the underlying resource when the k8s resource
  is deleted

This allows adopting externally managed resources into ACK for read-only
purposes like referencing values, without ACK attempting to manage the
resource lifecycle.

Signed-off-by: Amine Hilaly <hilalyamine@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/runtime/reconciler_test.go
Comment (a-hilaly) [inline pkg/runtime/reconciler_test.go:1]: More reasons to switch to envtest.. https://github.com/aws-controllers-k8s/community/issues/1321

**Review by michaelhtm — COMMENTED** [review]
🚀 🚀 
/lgtm

**Comment by knottnt** [comment]
Link to the feature request issue:
https://github.com/aws-controllers-k8s/community/issues/2008

**Pull request merged**